lib/kafo/hook_context.rb

```diff
@@ -196,5 +196,13 @@ def scenario_data
     def exit_code
       self.kafo.exit_code
     end
+
+    # Return the Puppet report, if any.
+    # Only available after Puppet actual ran.
+    #
+    # @return [Optional[Kafo::PuppetReport]]
+    def puppet_report
```

Comment:
If you try to call this in any of the pre-hooks, I assume it will just return `nil` to indicate there is no report yet? Whereas in any post hook this would be populated by the most recent report?

Reply:
That is indeed what I would think, though I didn't test it explicitly.